spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -749,6 +750,19 @@ public static TableIdentifier identifierToTableIdentifier(Identifier identifier)
     return TableIdentifier.of(Namespace.of(identifier.namespace()), identifier.name());
   }
 
+  public static String quotedFullIdentifier(String catalogName, Identifier identifier) {
+    List<String> parts =
+        ImmutableList.<String>builder()
+            .add(catalogName)
+            .addAll(Arrays.asList(identifier.namespace()))
+            .add(identifier.name())
+            .build();
+
+    return CatalogV2Implicits.MultipartIdentifierHelper(
+        JavaConverters.asScalaIteratorConverter(parts.iterator()).asScala().toSeq()
+    ).quoted();
+  }
```

Comment:
Using the `CatalogV2Implicits.MultipartIdentifierHelper` might be overkill.

On one hand, I like it as it's the canonical way of quoting. On the other hand, the definition looks really fancy bc it's extra scala-like, but it basically bolils down to the [existing quoted name function in SparkSchemaUtil#indexQuotedNameById ](https://github.com/apache/iceberg/blob/a5efb53f3588306731afed71ad422891a588ccdf/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSchemaUtil.java#L344)

Which is basically in its final form:
```python
'.'.join([ quoteIfNeeded(part) for part in parts ])
```

Reply:
``` java
private String quoteName(String name) {
    String escapedName = name.replace("`", "``"); // if the column name contains ` then escape it with another `
    return "`" + escapedName + "`";
  }
```

But the more I think about it, the more I'm actually comfortable with using the V2 implicits. So I'll leave that up to you, but thought I'd call out that the somewhat crazy usage of Scala within Java boils down to something not that complex.